service-runner.js

```diff
@@ -178,19 +204,35 @@ ServiceRunner.prototype._runMaster = function() {
     // Set up rolling restarts
     process.on('SIGHUP', this._rollingRestart.bind(this));
 
-    return this._startWorkers(this.config.num_workers);
+    return this._startWorkers(this.config.num_workers)
+    .then(function(workers) {
+        setTimeout(function() {
+            self._checkHeartbeat();
+        }, 1000);
+        return workers;
+    });
 };
 
-ServiceRunner.prototype._stopWorker = function(id) {
-    var worker = cluster.workers[id];
+ServiceRunner.prototype._stopWorker = function(workerId) {
+    var self = this;
+    var worker = cluster.workers[workerId];
+    self.workerHeartbeatTime[worker.process.pid] = {
```

Comment:
This is checked in here before we call this func: https://github.com/wikimedia/service-runner/pull/44/files#diff-b3efe45b2faf7882e08bbdaec207491dR156